modules/core/src/main/java/org/apache/ignite/internal/tostring/IgniteLogThrottle.java

```diff
@@ -0,0 +1,532 @@
+     * @param msg Message.
+     */
+    public static void debug(IgniteLogger log, String msg) {
+        debug(log, msg, (Throwable) null, OBJECT_EMPTY_ARRAY);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param msg Message.
+     * @param e Error (optional).
+     */
+    public static void debug(IgniteLogger log, String msg, @Nullable Throwable e) {
+        debug(log, msg, e, OBJECT_EMPTY_ARRAY);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param msg Message pattern.
+     * @param params List of arguments to be substituted in place of formatting anchors.
+     */
+    public static void debug(IgniteLogger log, String msg, Object... params) {
+        debug(log, msg, (Throwable) null, params);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param e Error (optional).
+     * @param msg Message pattern.
+     * @param params List of arguments to be substituted in place of formatting anchors.
+     */
+    public static void debug(IgniteLogger log, String msg, @Nullable Throwable e, Object... params) {
+        log(log, e, null, msg, LogLevel.DEBUG, params);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param throttleKey Messages with the same key will be throttled.
+     * @param msg Message.
+     */
+    public static void debug(IgniteLogger log, String throttleKey, String msg) {
+        debug(log, throttleKey, msg, null, OBJECT_EMPTY_ARRAY);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param throttleKey Messages with the same key will be throttled.
+     * @param e Error (optional).
+     * @param msg Message.
+     */
+    public static void debug(IgniteLogger log, String throttleKey, String msg, @Nullable Throwable e) {
+        debug(log, throttleKey, msg, e, OBJECT_EMPTY_ARRAY);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param throttleKey Messages with the same key will be throttled.
+     * @param msg Message pattern.
+     * @param params List of arguments to be substituted in place of formatting anchors.
+     */
+    public static void debug(IgniteLogger log, String throttleKey, String msg, Object... params) {
+        debug(log, throttleKey, msg, null, params);
+    }
+
+    /**
+     * Logs debug if needed.
+     *
+     * @param log Logger.
+     * @param throttleKey Messages with the same key will be throttled.
+     * @param msg Message pattern.
+     * @param e Error (optional).
+     * @param params List of arguments to be substituted in place of formatting anchors.
+     */
+    public static void debug(IgniteLogger log, String throttleKey, String msg, @Nullable Throwable e, Object... params) {
+        log(log, e, throttleKey, msg, LogLevel.DEBUG, params);
+    }
+
+    /** Clears cached messages. */
+    public static void clean() {
```

Comment:
Also, should we call it `clear` instead of `clean`?

Reply:
There is no longer a method.